File: ObjectRequestBrokerCS/ObjectRequestBrokerCS/vson/writers/FieldWriter.cs
using ObjectRequestBrokerCS.vson;
using System;
using System.Reflection;
using System.Text;

namespace ObjectRequestBrokerCS.vson.writers
{


    //JAVA TO C# CONVERTER TODO TASK: This Java 'import static' statement cannot be converted to C#:
    //	import static vson.FieldUtils.isPrimitive;

    public static class FieldWriter
    {

        public static string toJson(FieldInfo f, object @object)
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("\"").Append(f.Name).Append("\"").Append(": ");

            try
            {
                sb.Append(writeFieldValue(f.GetValue(@object)));
            }
            catch (Exception e)
            {
                Console.WriteLine(e.ToString());
                Console.Write(e.StackTrace);
            }

            return sb.ToString();
        }


        public static string writeFieldValue(object @object)
        {
            StringBuilder sb = new StringBuilder();

            if (FieldUtils.isPrimitive(@object.GetType()))
            { // attribute is primitive
                sb.Append(@object.ToString().ToLower());
                // using ToLower in order to transform boolean values "True" and "False" into 
                // the lower case "true" and "false" expected by JObject.Parse()
            }
            else if (@object is string)
            { // attribute is string
                sb.Append("\"").Append(@object.ToString()).Append("\"");

            }
            else if (@object.GetType().IsArray)
            { // attribute is array

                sb.Append("[\n");

                for (var i = 0; i < ((object[])@object).Length; i++)
                {
                    sb.Append(writeFieldValue(((object[])@object)[i])).Append(",\n");
                }
                // delete last comma
                sb.Remove(sb.Length - 1, 1).Append("]");

            }
            else // attribute is an Object
            {
                sb.Append(ObjectWriter.toJson(@object));
            }

            return sb.ToString();
        }


    }
}

File: ObjectRequestBrokerJava/src/main/java/ORB/vson/writers/FieldWriter.java
package ORB.vson.writers;


import java.lang.reflect.Array;
import java.lang.reflect.Field;

import static ORB.vson.FieldUtils.isPrimitive;

public class FieldWriter {

    public FieldWriter() {
    }


    public static String toJson(Field f, Object object) {
        StringBuffer sb = new StringBuffer();

        sb.append("\"").append(f.getName()).append("\"").append(": ");

        try {
            sb.append(writeFieldValue(f.get(object)));
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }

        return sb.toString();
    }


    public static String writeFieldValue(Object object) {
        StringBuilder sb = new StringBuilder();

        if (isPrimitive(object.getClass())) { // attribute is primitive
            sb.append(String.valueOf(object));

        }else if (object.getClass().equals(String.class)) { // attribute is string
            sb.append("\"").append(String.valueOf(object)).append("\"");

        } else if (object.getClass().isArray()) { // attribute is array

            sb.append("[\n");

            for (int i = 0; i < Array.getLength(object); i++)
                sb.append(writeFieldValue(Array.get(object, i))).append(",\n");
            // delete last comma
            sb.replace(sb.length() - 2, sb.length() - 1, "").append("]");

        } else   // attribute is an Object
            sb.append(ObjectWriter.toJson(object));

        return sb.toString();
    }


}
